bundler/lib/bundler/definition.rb

```diff
@@ -56,7 +57,19 @@ def self.build(gemfile, lockfile, unlock)
     #   to be updated or true if all gems should be updated
     # @param ruby_version [Bundler::RubyVersion, nil] Requested Ruby Version
     # @param optional_groups [Array(String)] A list of optional groups
-    def initialize(lockfile, dependencies, sources, unlock, ruby_version = nil, optional_groups = [], gemfiles = [])
+    # @param lockfile_contents [String, nil] The contents of the lockfile
+    # @param plugin_dependencies [Array(Bundler::Dependency)] array of plugin dependencies from Gemfile
+    # @param plugin_sources [Bundler::SourceList]
+    def initialize(lockfile,
+      dependencies,
+      sources,
+      unlock,
+      ruby_version = nil,
+      optional_groups = [],
+      gemfiles = [],
+      lockfile_contents = nil,
```

Comment:
I wonder if we could treat `plugins` just as regular dependencies, and include them in the lockfile normally. Maybe that would work to simplify this and avoid having to generate an "in memory lockfile" every time? We could treat them like we treat `gemspec` dependencies, with a special `type:` value, so that we can implement the additional behavior that plugins have associated to them. Does that make sense? 🤔